frontend/src/components/Dropdown/components/SelectPopover/SelectPopover.tsx

```diff
@@ -1,48 +1,51 @@
-import { FC, useRef, useState } from 'react'
-import { Modifier, usePopper } from 'react-popper'
+import { FC, useLayoutEffect, useMemo, useRef } from 'react'
 import { Box, useMergeRefs, useOutsideClick } from '@chakra-ui/react'
+import {
+  autoUpdate,
+  flip,
+  hide,
+  offset,
+  useFloating,
+} from '@floating-ui/react-dom-interactions'
 
 import { useSelectContext } from '~components/Dropdown/SelectContext'
 
 import { SelectPopoverContext } from './SelectPopoverContext'
 
-export const matchWidth: Modifier<'matchWidth'> = {
-  name: 'matchWidth',
-  enabled: true,
-  phase: 'beforeWrite',
-  requires: ['computeStyles'],
-  fn: ({ state }) => {
-    state.styles.popper.width = `${state.rects.reference.width}px`
-  },
-  effect:
-    ({ state }) =>
-    () => {
-      const reference = state.elements.reference as HTMLElement
-      state.elements.popper.style.width = `${reference.offsetWidth}px`
-    },
-}
-
 export const SelectPopoverProvider: FC = ({ children }): JSX.Element => {
-  const [referenceElement, setReferenceElement] =
-    useState<HTMLDivElement | null>(null)
-  const [popperElement, setPopperElement] = useState<HTMLDivElement | null>(
-    null,
-  )
+  const { styles, setIsFocused, isOpen } = useSelectContext()
+
+  const wrapperRef = useRef<HTMLDivElement | null>(null)
 
-  const {
-    styles: popperStyles,
-    attributes: popperAttributes,
-    update,
-  } = usePopper(referenceElement, popperElement, {
+  const { x, y, refs, reference, floating, strategy, update } = useFloating({
     placement: 'bottom-start',
-    strategy: 'fixed',
-    modifiers: [matchWidth],
+    strategy: 'absolute',
+    open: isOpen,
+    middleware: [
+      // offset middleware should be the first middleware
+      offset(1),
+      flip(),
+      hide(),
+    ],
   })
 
-  const { styles, setIsFocused, isOpen } = useSelectContext()
+  const mergedReferenceRefs = useMergeRefs(reference, wrapperRef)
 
-  const wrapperRef = useRef<HTMLDivElement | null>(null)
-  const mergedPopperRefs = useMergeRefs(setReferenceElement, wrapperRef)
+  const floatingStyles = useMemo(
+    () => ({
+      position: strategy,
+      top: y ?? 0,
+      left: x ?? 0,
+    }),
+    [strategy, x, y],
+  )
+
+  // Allows
+  useLayoutEffect(() => {
+    if (isOpen && refs.reference.current && refs.floating.current) {
+      return autoUpdate(refs.reference.current, refs.floating.current, update)
+    }
+  }, [isOpen, update, refs.floating, refs.reference])
```

Comment:
/justNoting: I don't have a suggestion here, but reading this file on its own, it's a little hard to keep track what all the refs are about 😰 :
* `reference`
* `wrapperRef`
* `mergedReferenceRefs`
* `refs.reference`
* `refs.floating`

Reading [the doc](https://floating-ui.com/docs/react-dom) helped a bit!